### DISCUSSION TITLE
infrastructure/proxmox/gpu-passthrough/pgu-passthrough-to-vm/

### DISCUSSION THREAD

**Discussion by giscus** [fire1ce/3os.org#67] [General]
# infrastructure/proxmox/gpu-passthrough/pgu-passthrough-to-vm/

Vestibulum quam turpis, lacinia lacinia ex finibus, egestas malesuada nunc. Maecenas euismod neque rhoncus suscipit viverra. Nulla venenatis enim vel mauris ornare viverra.

https://3os.org/infrastructure/proxmox/gpu-passthrough/pgu-passthrough-to-vm/

**Comment by GaNiziolek** [comment]
Thanks for the great content, this site is perfect for beginners as well as experts.
I intend to contribute with other posts on github as soon as possible!

**Reply by fire1ce** [reply]
@GaNiziolek You are welcomed! 
Contribution is always welcomed, it's not easy to be a single maintainer.

**Comment by fengcda** [comment]
Thanks for the guide, was a helpful jumping point to getting my RX 580 working on kernel 5.15. The process was similar, but I didn't need as many changes in GRUB + an AMD exclusive package called https://github.com/gnif/vendor-reset. Then I added a scripthook command to call `echo 'device_specific' > /sys/bus/pci/devices/0000:09:00.0/reset_method` so passthrough continued to work on VM shutdown.

**Reply by mosflix** [reply]
Hi, would you be kind enough to share your process! e.g what steps you followed to get your AMD working. I've done everything I can and still unable to passthrough AMD GPU. It just goes straight to a black screen. I have the RX 580 as well.

**Comment by MichelMelhem** [comment]
I'm getting the error [  173.153160] vfio-pci 0000:0a:00.0: BAR 1: can't reserve [mem 0xd0000000-0xdfffffff 64bit pref] each time the vm start. I added all the arguments needed in grub. What could be the issue ?

**Comment by maitpatni** [comment]
Hello dose this work on Mobile Gpu Processors as well ?
it is detectinh 3d video driver in windows and i am not being abel to install any video driver
